packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -536,17 +590,52 @@ class MemoryController extends DisposableController
     return libraryRoot;
   }
 
+  void addAllToNode(
+      Reference root, Map<String, List<HeapGraphClassLive>> allItems) {
+    allItems.forEach((libraryName, classes) {
+      LibraryReference libReference = root.children.singleWhere((library) {
```

Comment:
this would be O(n^2) in terms of the #of libraries. We should probably add a set of all libraries so that this is O(n). That could matter for large projects with >4000 libraries.

Reply:
Yes I'll add a TODO to eliminate the Map and have a Set of libraries.